blocks/author-list/author-list.js

```diff
@@ -0,0 +1,133 @@
+    loadMoreButton.addEventListener('click', () => {
+      // get the next 50 authors
+      const nextAuthors = authors.slice(numInitialLodedAuthors);
+      nextAuthors.forEach((author) => {
+        const newBlock = createAuthorCardBlock(author);
+        block.append(newBlock);
+      });
+      loadMoreContainer.remove();
+    });
+    block.append(loadMoreContainer);
+  }
+  loadBlocks(block);
+}
+
+function createLoadMoreButton() {
+  const loadMoreContainer = document.createElement('div');
+  loadMoreContainer.classList.add('author-load-more-container');
+  const loadMoreButton = document.createElement('button');
+  loadMoreContainer.append(loadMoreButton);
+  loadMoreButton.classList.add('author-list-load-more-button');
+  loadMoreButton.textContent = 'Load more';
+  return { loadMoreButton, loadMoreContainer };
+}
+
+function buildAuthorListItem(className, content) {
+  const list = Array.isArray(content) ? content : [content];
+  const container = document.createElement('div');
+  container.classList.add(className);
+  list.forEach((val) => {
+    if (val) {
+      if (typeof val === 'string') {
+        container.innerHTML += val;
+      } else {
+        container.appendChild(val);
+      }
+    }
+  });
+  return (container);
+}
+
+function p(content) {
+  const result = document.createElement('p');
+  result.append(content);
+  return result;
+}
+
+/* convenience function to create a block from a JSON object from authors.json */
+function createAuthorCardBlock(author) {
+  const picture = createOptimizedPicture(author.image, 'author-image', true);
+  const heading = document.createElement('h3');
+  const anchor = document.createElement('a');
+  anchor.href = author.path;
+  anchor.textContent = author.name;
+  heading.append(anchor);
+
+  const authorLinkContainer = document.createElement('div');
+  authorLinkContainer.classList.add('author-links');
+  // parse the author.links string and iterate over links
+  addAuthorLinks(author, authorLinkContainer);
+
+  return buildAuthorListItem('author-list-item', [
+    p(picture),
+    heading,
+    p(author.description),
+    authorLinkContainer,
+  ]);
+}
+
+function addAuthorLinks(author, authorLinkContainer) {
+  if (!author.links) return;
+  let arr = JSON.parse(author.links);
+  if (arr.length > 0) {
+    arr = arr[0].split(',');
+  }
+  const socialLinksContainer = document.createElement('div');
+  socialLinksContainer.classList.add('social-links');
+  function addSocialLink(socialLink) {
+    const container = document.createElement('p');
+    container.appendChild(socialLink);
+    socialLinksContainer.appendChild(container);
+  }
+  arr.forEach((link) => {
+    const socialLink = document.createElement('a');
+    socialLink.href = link;
+    socialLink.target = '_blank';
+    if (link.includes('facebook')) {
+      socialLink.innerHTML = '<span class="icon icon-facebook"><svg xmlns="http://www.w3.org/2000/svg"><use href="#icons-sprite-facebook"></use></svg></span>';
```

Comment:
You can omit the svg code. decorateIcons() will do that for you

Reply:
fixed